luigi/task_register.py

```diff
@@ -135,17 +135,18 @@ def _get_reg(cls):
         # We return this in a topologically sorted list of inheritance: this is useful in some cases (#822)
         reg = OrderedDict()
         for cls in cls._reg:
-            name = cls.task_family
-
-            if name in reg and reg[name] != cls and \
-                    reg[name] != cls.AMBIGUOUS_CLASS and \
-                    not issubclass(cls, reg[name]):
-                # Registering two different classes - this means we can't instantiate them by name
-                # The only exception is if one class is a subclass of the other. In that case, we
-                # instantiate the most-derived class (this fixes some issues with decorator wrappers).
-                reg[name] = cls.AMBIGUOUS_CLASS
-            else:
-                reg[name] = cls
+            if cls._visible_in_registry:
+                name = cls.get_task_family()
+
+                if name in reg and reg[name] != cls and \
+                        reg[name] != cls.AMBIGUOUS_CLASS and \
```

Comment:
Great catch. I added a test case. for this. You were right, the current behavior is broken (see soon up and coming update to PR) :)